nengo/spa/thalamus.py

```diff
@@ -188,6 +189,8 @@ def add_route_effect(self,
             subdim = self.subdim_channel
             if isinstance(target_module, nengo.spa.Buffer):
                 subdim = target_module.state.dimensions_per_ensemble
+            elif isinstance(source_module, nengo.spa.Buffer):
+                subdim = source_module.state.dimensions_per_ensemble
```

Comment:
On a related note, what if it connects to ```spa.Memory```. Should there be logic for that here too?